quill-cassandra/src/main/scala/io/getquill/context/cassandra/CassandraSessionContext.scala

```diff
@@ -31,8 +32,8 @@ abstract class CassandraSessionContext[N <: NamingStrategy](
   override type RunActionReturningResult[T] = Unit
   override type RunBatchActionReturningResult[T] = Unit
 
-  private val preparedStatementCache =
-    new PrepareStatementCache(preparedStatementCacheSize)
+  private val asyncCache = new PrepareStatementCache[Future[PreparedStatement]](preparedStatementCacheSize)
+  private val syncCache = new PrepareStatementCache[PreparedStatement](preparedStatementCacheSize)
```

Comment:
Should we create both instances of sync and async caches for each context?
May be ```lazy val```?

Reply:
`lazy val` sounds fine. Yet IMO if empty, it does not make much difference. If non-empty both are going to be there anyway.